crates/subspace-networking/src/utils/piece_provider.rs

```diff
@@ -173,4 +173,39 @@ where
         })
         .await
     }
+
+    /// Get piece from a particular peer.
+    pub async fn get_piece_from_peer(
```

Comment:
Hm... we still have `get_piece` method for some reason as well as internal `get_piece_from_storage` that judging by description relies on announcements that we supposedly removed :confused: Did previous commit not clean up everything related to announcements? Because I'd expect so.

Also this whole `PieceProvider` seems to be somewhat useless without access to piece filters. Either those filters need to move into `subspace-networking` (which I think wouldn't be nice) or this needs to be moved out, since this is just a simple wrapper around generic request with optional validation.